app/javascript/widget/App.vue

```diff
@@ -80,6 +122,37 @@ export default {
         Vue.config.lang = locale;
       }
     },
+    setPosition(position) {
+      const widgetPosition = position || 'right';
+      this.widgetPosition = widgetPosition;
+    },
+    registerUnreadEvents() {
+      bus.$on('on-agent-message-recieved', () => {
+        this.setUnreadView();
+      });
+      bus.$on('on-unread-view-clicked', () => {
+        this.unsetUnreadView();
+      });
```

Comment:
Can we do it without event bus?

Reply:
event bus is part of vue and it is the best suited here rather than depending on callbacks into action cable listner.